### PR TITLE
Paginated delete (fixes #987)

### LINKED ISSUE #987: Delete is not affected by limit query
The [current documentation](https://kinto.readthedocs.io/en/latest/api/1.x/pagination.html) says:

> Plural endpoints support limiting the number of elements returned. For example, you can retrieve a fixed number of records in a collection. To do this, provide a `_limit` query parameter specifying the number of records to return.
(...)
Pagination works on any plural endpoint.

It doesn't work on delete endpoints and I'm not sure if it should. I guess the behavior in this case might be a bit dubious. Would it limit the number of deleted entries or the number of returned deleted entries? 

An example:
```json
$ http delete localhost:8888/v1/buckets?_limit=1 -a a:HTTP/1.1 200 OK
Access-Control-Expose-Headers: Retry-After, Content-Length, Alert, Backoff
Content-Length: 181
Content-Type: application/json; charset=UTF-8
Date: Sun, 18 Dec 2016 18:53:46 GMT
Server: waitress
X-Content-Type-Options: nosniff

{
    "data": [
        {
            "deleted": true,
            "id": "b1",
            "last_modified": 1482087226574
        },
        {
            "deleted": true,
            "id": "b2",
            "last_modified": 1482087226575
        },
        {
            "deleted": true,
            "id": "b3",
            "last_modified": 1482087226576
        }
    ]
}


```

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #987 

- [x] Add limit/sorting to querystring
- [x] Add next-page header to plural delete responses
- [x] Plug model resource models
- [x] Add necessary parameters to storage backend
- [x] Check out documentation

### CODE REVIEW

File: kinto/core/resource/model.py [hunk 1]
```diff
@@ -110,6 +111,12 @@ def delete_records(self, filters=None, parent_id=None):
             comparison (see `kinto.core.utils.COMPARISON`). All filters
             are combined using *AND*.
         :type filters: list of :class:`kinto.core.storage.Filter`
+        :param sorting: XXX
```
Comment (Natim) [inline kinto/core/resource/model.py]: :param filters: Optionally filter the records by their attribute.
             Each filter in this list is a tuple of a field, a value and a
             comparison (see `kinto.core.utils.COMPARISON`). All filters
             are combined using *AND*.
         :type filters: list of :class:`kinto.core.storage.Filter`
 
         :param sorting: Optionnally sort the records by attribute.
             Each sort instruction in this list refers to a field and a
             direction (negative means descending). All sort instructions are
             cumulative.
         :type sorting: list of :class:`kinto.core.storage.Sort`
 
         :param pagination_rules: Optionnally paginate the list of records.
             This list of rules aims to reduce the set of records to the current
             page. A rule is a list of filters (see `filters` parameter),
             and all rules are combined using *OR*.
         :type pagination_rules: list of list of
             :class:`kinto.core.storage.Filter`
 
         :param int limit: Optionnally limit the number of records to be
            retrieved.

File: kinto/core/resource/model.py [hunk 2]
```diff
@@ -111,6 +112,22 @@ def delete_records(self, filters=None, parent_id=None):
             are combined using *AND*.
         :type filters: list of :class:`kinto.core.storage.Filter`
 
+        :param sorting: Optionnally sort the records by attribute.
```
Comment (glasserc) [inline kinto/core/resource/model.py]: `Optionally` 🐑

**Comment by leplatrem** [comment]
@gabisurita so it should return the total records that will deleted accross all pages right?

**Comment by gabisurita** [comment]
@leplatrem Yes, I think so. It should return the number of buckets that would be deleted without pagination.

**Comment by leplatrem** [comment]
could you open an issue please :) ?
I think it's trivial because we do a get_all() just before performing the deletion...

**Pull request merged**